src/stdio_mgr/types.py

```diff
@@ -0,0 +1,179 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+import abc
+import sys
+from contextlib import ExitStack, suppress
+from io import TextIOBase
+
+import _collections_abc
+
+# AbstractContextManager was introduced in Python 3.6
+# and may be used with typing.ContextManager.
+try:
+    from contextlib import AbstractContextManager
+except ImportError:  # pragma: no cover
+
+    class AbstractContextManager(abc.ABC):
```

Comment:
this is needed because now all of the `__enter__`/`__exit__` use `super()` so that changes in the MRO always have the desired effect, and `object` doent have `__enter__`/`__exit__`.

Reply:
I need to look at this further, to understand it, before merging -- doing just a quick partial review for now

Reply:
It still has coverage problems, sort-of-intentionally, as I'd rather not have it merged until `TextIOTuple` is integrated or removed, and I'd like to have https://github.com/bskinn/stdio-mgr/pull/78 or https://github.com/bskinn/stdio-mgr/pull/64 also mostly complete.  It is  separate PR as it is common to both paths forward.

Reply:
c.f. https://github.com/jazzband/contextlib2/pull/21 for the backport of AbstractContextManager specifically.  I suspect I can get a competent review there quite quickly by poking the right people.

Reply:
Would there be value in a `GenericIOTuple` in addition to, or in place of, the `Fake IOTuple`?

The former is for streams with some minimal common set of real IO functionality, whereas the latter is for things that specifically are nonfunctional for IO in some way(s)?

Too early to tell?